File: NewDwarfControl/src/main/webapp/js/controllers/lancamentosController.js
angular.module('Dwarf_Control').controller('lancamentosController', function($scope, contasService, lancamentosService){

	//Preenchimento dos campos da data com a data de hj.
	$scope.requestLancamentos = {};
	$scope.requestLancamentos.dataInicial =  new Date();
	$scope.requestLancamentos.dataFinal =  new Date();


	$scope.menssagem = {};
	$scope.saldoAnterior = 0;
	$scope.saldoTotal = 0;
	$scope.exclusaoLancamento = false;
	$scope.editarLancamento = {};

	$scope.contasUsuario = function(){

		$scope.exibirCarregando = true;
		$scope.menssagem.erro = false;
		$scope.menssagem.sucesso = false;

		contasService.getContas().then(function mySuccess(response) {
			//Ação Caso Sucesso
			$scope.exibirCarregando = false;
			$scope.listaContas = response.data;

		}, function myError(response) {
			//Ação caso Erro
			$scope.exibirCarregando = false;
			$scope.menssagem.erro = true;
			$scope.menssagem.descricao = "Erro interno de processamento no Servidor!";
			
		});

	};

	//Carregando Contas - To usando Funlções já pontra no Serviço do Java uploadCSV
	$scope.contasUsuario();

	//Método responsável por obter um saldo anterior no Back-End conforme request.
	$scope.obterSaldoAnterior = function(request){

		$scope.exibirCarregando = true;
		$scope.menssagem.erro = false;
		$scope.menssagem.sucesso = false;

		lancamentosService.obterSaldoAnterior(request).then(function mySuccess(response) {
			//Ação Caso Sucesso
			$scope.exibirCarregando = false;
			//Ação  da consulta.
			$scope.saldoAnterior = response.data;

		}, function myError(response) {
			//Ação caso Erro
			$scope.exibirCarregando = false;
			$scope.menssagem.erro = true;
			$scope.menssagem.descricao = "Erro interno de processamento no Servidor!";
			
		});

	}

	//Método responsável por obter um saldo Total do Periodo no Back-End conforme request.
	$scope.obterSaldoTotal = function(request){

		$scope.exibirCarregando = true;
		$scope.menssagem.erro = false;
		$scope.menssagem.sucesso = false;

		lancamentosService.obterSaldoTotalPeriodo(request).then(function mySuccess(response) {
			//Ação Caso Sucesso
			$scope.exibirCarregando = false;
			//Ação  da consulta.
			$scope.saldoTotal = response.data;

		}, function myError(response) {
			//Ação caso Erro
			$scope.exibirCarregando = false;
			$scope.menssagem.erro = true;
			$scope.menssagem.descricao = "Erro interno de processamento no Servidor!";
			
		});

	}

	//Obter lancamentos por data incial final e conta
	$scope.obterLancamentos = function(request){

		$scope.exibirCarregando = true;
		$scope.menssagem.erro = false;
		$scope.menssagem.sucesso = false;

		lancamentosService.obterLancamentos(request).then(function mySuccess(response) {
			//Ação Caso Sucesso
			$scope.exibirCarregando = false;
			//Ação  da consulta.
			$scope.listaLancamentos = response.data;

			//Logo após está ação preciso obter o Saldo Anterior e o Saldo Total
			$scope.obterSaldoAnterior(request);
			$scope.obterSaldoTotal(request);


		}, function myError(response) {
			//Ação caso Erro
			$scope.exibirCarregando = false;
			$scope.menssagem.erro = true;
			$scope.menssagem.descricao = "Erro interno de processamento no Servidor!";
			
		});

	};



	//Função para realizar a deleção do lanaçamento no Back-End
	$scope.deleteLancamento = function(id){

		$scope.exibirCarregando = true;
		$scope.menssagem.erro = false;
		$scope.menssagem.sucesso = false;

		lancamentosService.deleteLancamento(id).then(function mySuccess(response) {
			//Ação Caso Sucesso
			$scope.exibirCarregando = false;
			//Ação  da consulta.
			$scope.exclusaoLancamento = response.data;

			if($scope.exclusaoLancamento == true){

				//Logo após deleção preciso atualizar o grid vou usar o que está setado no bind lá no Front-End
				$scope.obterLancamentos($scope.requestLancamentos);
				$scope.obterSaldoAnterior($scope.requestLancamentos);
				$scope.obterSaldoTotal($scope.requestLancamentos);

			} else {
				//Ação caso Erro
				$scope.exibirCarregando = false;
				$scope.menssagem.erro = true;
				$scope.menssagem.descricao = "Erro ao tentar excluir lanaçamento!";
			}

		}, function myError(response) {
			//Ação caso Erro
			$scope.exibirCarregando = false;
			$scope.menssagem.erro = true;
			$scope.menssagem.descricao = "Erro interno de processamento no Servidor!";
			
		});
	};



	//Método responsável por cadastrar novos lançamentos
	$scope.cadastrarLancamento = function(novoLancamento){

		//Escondendo caixas de menssagem e exibindo carregando
		$scope.exibirCarregando = true;
		$scope.menssagem.erro = false;
		$scope.menssagem.sucesso = false;

		lancamentosService.novoLancamento(novoLancamento).then(function mySuccess(response) {
			
			//Ação Caso Sucesso
			$scope.exibirCarregando = false;
			$scope.resultadoNovoLancamento = response.data;

			if($scope.resultadoNovoLancamento){

				$scope.menssagem.sucesso = true;
				$scope.menssagem.descricao = "Lançamento cadastrado com sucesso!"

				//Limpando campo, de cadastro
				delete $scope.novoLancamento;

				//Atualizando a lista exibida para exibir no mesmo intervalo e conta do lançamento cadastrado
				$scope.requestLancamentos.conta = novoLancamento.conta;
				$scope.requestLancamentos.dataInicial = novoLancamento.data;
				$scope.requestLancamentos.dataFinal = novoLancamento.data;

				$scope.obterLancamentos($scope.requestLancamentos);
				$scope.obterSaldoAnterior($scope.requestLancamentos);
				$scope.obterSaldoTotal($scope.requestLancamentos);


			}
			
		}, function myError(response) {
			//Ação caso Erro
			$scope.exibirCarregando = false;
			$scope.menssagem.erro = true;
			$scope.menssagem.descricao = "Erro interno de processamento no Servidor!";
			
		});


	};


	//Método responsável por obter o lançamento desejado para edição e preencher a tela de edição
	$scope.obterLancamento = function(id){

		$scope.exibirCarregando = true;
		$scope.menssagem.erro = false;
		$scope.menssagem.sucesso = false;

		lancamentosService.obterLancamento(id).then(function mySuccess(response) {
			//Ação Caso Sucesso
			$scope.exibirCarregando = false;
			$scope.editarLancamento.id = response.data.id;
			$scope.editarLancamento.conta = response.data.conta;
			$scope.editarLancamento.valor = response.data.Valor;
			$scope.editarLancamento.descricao = response.data.descricao;

		}, function myError(response) {
			//Ação caso Erro
			$scope.exibirCarregando = false;
			$scope.menssagem.erro = true;
			$scope.menssagem.descricao = "Erro interno de processamento no Servidor!";
			
		});

	};


	//Metodo responsável por atualizar o lancamento e exibir
	$scope.atualizarLancamento = function(lancamento){

		$scope.exibirCarregando = true;
		$scope.menssagem.erro = false;
		$scope.menssagem.sucesso = false;

		lancamentosService.updateLancamento(lancamento).then(function mySuccess(response) {
			//Ação Caso Sucesso
			$scope.exibirCarregando = false;


			if(response.data){

				//Atualizando a lista exibida para exibir no mesmo intervalo e conta do lançamento cadastrado
				$scope.requestLancamentos.conta = lancamento.conta;
				$scope.requestLancamentos.dataInicial = lancamento.data;
				$scope.requestLancamentos.dataFinal = lancamento.data;

				$scope.obterLancamentos($scope.requestLancamentos);
				$scope.obterSaldoAnterior($scope.requestLancamentos);
				$scope.obterSaldoTotal($scope.requestLancamentos);


			}

		}, function myError(response) {
			//Ação caso Erro
			$scope.exibirCarregando = false;
			$scope.menssagem.erro = true;
			$scope.menssagem.descricao = "Erro interno de processamento no Servidor!";
			
		});

	};
	
});

File: NewDwarfControl/src/main/java/br/com/controllers/LancamentosController.java
package br.com.controllers;


import java.util.List;

import javax.servlet.http.HttpSession;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import br.com.entitys.Lancamento;
import br.com.models.LancamentoDTO;
import br.com.models.RequestLancamentosDTO;
import br.com.services.LancamentosService;

@RestController
public class LancamentosController {
			
	private LancamentosService lancamentosService = new LancamentosService();
	
	/**
	 * Método responsável por entregar os lançamentos conforme o periodo
	 * e conta enviados em uma request.
	 * 
	 * @param request
	 * @return
	 */
	@RequestMapping(value="/get/lancamentos", method=RequestMethod.POST)
	public ResponseEntity<List<Lancamento>> obterLancamentos(@RequestBody RequestLancamentosDTO request) {
		
		List<Lancamento> listaLancamentos = lancamentosService.obterLancamentos(request);
		
		if(listaLancamentos.size() == 0) {
			
			return new ResponseEntity<List<Lancamento>>(HttpStatus.NO_CONTENT);
			
		} else {
			
			return new ResponseEntity<List<Lancamento>>(listaLancamentos, HttpStatus.OK);
			
		}
	}
	
	
	/**
	 * Método responsável por entregar o saldo anterior conforme o periodo
	 * e conta enviados em uma request
	 * 
	 * @param requestLancamentos
	 * @return
	 */
	@RequestMapping(value="/get/saldoAnterior", method=RequestMethod.POST)
	public ResponseEntity<Float> getSaldoAnterior(@RequestBody RequestLancamentosDTO requestLancamentos){
		
		Float retorno = lancamentosService.getSaldoAnterior(requestLancamentos);
		
		return new ResponseEntity<Float>(retorno, HttpStatus.OK);
		
	}	
	
	
	/**
	 * Método responsável por entregar o saldo total conforme o periodo (dataFinal)
	 * e conta enviados em uma request
	 * 
	 * @param requestLancamentos
	 * @return
	 */
	@RequestMapping(value="/get/saldoTotalPeriodo", method=RequestMethod.POST)
	public ResponseEntity<Float> getSaldoTotalPeriodo(@RequestBody RequestLancamentosDTO requestLancamentos){
		
		Float retorno = lancamentosService.obterSaldoTotalAteFimPeriodo(requestLancamentos);
		
		return new ResponseEntity<Float>(retorno, HttpStatus.OK);
		
	}
	
	/***
	 * Método responsável por excluir um lançamento. É
	 * necessário antes de excluir veriricar se o lançamento pertence
	 * ao usuário logado!
	 * 
	 * @param idLancamento
	 * @return
	 */
	@RequestMapping(value="/delete/lancamento/{id}", method=RequestMethod.DELETE)
	private Boolean excluirLancamento(@PathVariable("id") int idLancamento, HttpSession session) {
		
		//Chamando método competente
		return lancamentosService.excluirLancamento(idLancamento, session);
		
	}
	
	
	/***
	 * Método responsável por cadastrar novos lançamentos. 
	 * 
	 * @param novoLancamento novo lançamento a ser cadastrado.
	 * @return
	 */
	@RequestMapping(value="/new/lancamento", method=RequestMethod.POST)
	private Boolean novoLancamento(@RequestBody LancamentoDTO novoLancamento) {
		
		return lancamentosService.cadastrarLancamento(novoLancamento.toLancamento(false));
		
	}
	
	
	/***
	 * Método responsável por obter os dados de lançamento para disponibilizar para 
	 * edição.
	 * 
	 * @param idLancamento
	 * @param session
	 * @return
	 */
	@RequestMapping(value="/get/lancamento/{id}", method=RequestMethod.GET)
	private Lancamento getLancamento(@PathVariable("id") int idLancamento, HttpSession session) {
		
		//Chamando método competente
		return lancamentosService.getLancamento(idLancamento, session);
		
	}
	
	
	/**
	 * Serviço esposoto para atualizar Lancamento
	 * 
	 * @param lancamento
	 * @return
	 */
	@RequestMapping(value="/update/lancamento", method=RequestMethod.POST)
	private Boolean updateLancamento(@RequestBody LancamentoDTO lancamento) {
		
		return lancamentosService.updateLancamento(lancamento.toLancamento(true));
		
	}
	
}